iotdb/src/main/java/org/apache/iotdb/db/qp/constant/DatetimeUtils.java

```diff
@@ -153,59 +298,185 @@ private DatetimeUtils() {
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SLASH_NS)
+
       /**
        * such as '2011.12.03T10:15:30+01:00' or '2011.12.03T10:15:30.123+01:00'.
        */
       .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT)
+
+      /**
+       * such as '2011.12.03T10:15:30+01:00' or '2011.12.03T10:15:30.123456+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_US)
+
+      /**
+       * such as '2011.12.03T10:15:30+01:00' or '2011.12.03T10:15:30.123456789+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_NS)
+
       /**
        * such as '2011-12-03 10:15:30+01:00' or '2011-12-03 10:15:30.123+01:00'.
        */
       .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SPACE)
+
+      /**
+       * such as '2011-12-03 10:15:30+01:00' or '2011-12-03 10:15:30.123456+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SPACE_US)
+
+      /**
+       * such as '2011-12-03 10:15:30+01:00' or '2011-12-03 10:15:30.123456789+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SPACE_NS)
+
       /**
        * such as '2011/12/03 10:15:30+01:00' or '2011/12/03 10:15:30.123+01:00'.
        */
       .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SLASH_WITH_SPACE)
+
+      /**
+       * such as '2011/12/03 10:15:30+01:00' or '2011/12/03 10:15:30.123456+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SLASH_WITH_SPACE_US)
+
+      /**
+       * such as '2011/12/03 10:15:30+01:00' or '2011/12/03 10:15:30.123456789+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_SLASH_WITH_SPACE_NS)
+
       /**
        * such as '2011.12.03 10:15:30+01:00' or '2011.12.03 10:15:30.123+01:00'.
        */
-      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_WITH_SPACE).toFormatter();
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_WITH_SPACE)
+
+      /**
+       * such as '2011.12.03 10:15:30+01:00' or '2011.12.03 10:15:30.123456+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_WITH_SPACE_US)
+
+      /**
+       * such as '2011.12.03 10:15:30+01:00' or '2011.12.03 10:15:30.123456789+01:00'.
+       */
+      .appendOptional(ISO_OFFSET_DATE_TIME_WITH_DOT_WITH_SPACE_NS).toFormatter();
 
   public static long convertDatetimeStrToMillisecond(String str, ZoneId zoneId)
       throws LogicalOperatorException {
     return convertDatetimeStrToMillisecond(str, toZoneOffset(zoneId), 0);
   }
 
+  public static long getInstantWithPrecision(String str, String timestampPrecision,
+      TSFileConfig.PrecisionType precisionType) {
+    /* split str with dot */
+    if (str.substring(20) != "") {
+      char[] st = str.substring(20).toCharArray();
+      for (char s : st) {
+        if (s == '.') {
+          if (str.substring(20).length() != precisionType.getDigit() && timestampPrecision
+              .equals(precisionType.getName())) {
+            System.out.println(str.substring(20));
+            System.exit(-1);
```

Comment:
do not use system.exit(). If system.exit(), then a single error insertion will kill Server.  You should throw exception and handle exception outside